test/integration/scheduler/predicates_test.go

```diff
@@ -1150,6 +1155,90 @@ func TestEvenPodsSpreadPredicate(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			allPods := append(tt.existingPods, tt.incomingPod)
 			defer testutils.CleanupPods(cs, t, allPods)
+
+			for _, pod := range tt.existingPods {
+				createdPod, err := cs.CoreV1().Pods(pod.Namespace).Create(context.TODO(), pod, metav1.CreateOptions{})
+				if err != nil {
+					t.Fatalf("Error while creating pod during test: %v", err)
+				}
+				err = wait.Poll(pollInterval, wait.ForeverTestTimeout, testutils.PodScheduled(cs, createdPod.Namespace, createdPod.Name))
+				if err != nil {
+					t.Errorf("Error while waiting for pod during test: %v", err)
+				}
+			}
+			testPod, err := cs.CoreV1().Pods(tt.incomingPod.Namespace).Create(context.TODO(), tt.incomingPod, metav1.CreateOptions{})
+			if err != nil && !apierrors.IsInvalid(err) {
+				t.Fatalf("Error while creating pod during test: %v", err)
+			}
+
+			if tt.fits {
+				err = wait.Poll(pollInterval, wait.ForeverTestTimeout, podScheduledIn(cs, testPod.Namespace, testPod.Name, tt.candidateNodes))
+			} else {
+				err = wait.Poll(pollInterval, wait.ForeverTestTimeout, podUnschedulable(cs, testPod.Namespace, testPod.Name))
+			}
+			if err != nil {
+				t.Errorf("Test Failed: %v", err)
+			}
+		})
+	}
+}
+
+// TestEvenPodsSpreadPredicate verifies that EvenPodsSpread predicate functions well.
+func TestPodTopologySpreadMinDomainsPredicate(t *testing.T) {
```

Comment:
Any reason why not add these cases in the existing `Test`?

Reply:
It'd be easier to have the existing `TestEvenPodsSpreadPredicate` piggyback the new tests.

BTW: could you help rename `TestEvenPodsSpreadPredicate` to `TestPodTopologySpreadFilter`?

Reply:
> So, if we want to enable MinDomains, we need to run defer featuregatetesting.SetFeatureGateDuringTest(...)() before running initTest fn to enable MinDomains feature just like at a little lower from here.

You can add a boolean in the case struct to enable/disable the feature gate.

> On the other hand, if we add the test case for MinDomains to existing TestEvenPodsSpreadPredicate, we may need to restart the scheduler in each test case.

We should. That's a good practice.

Reply:
Okay, I fixed that way.